File: Testing/node.py

class Node:

    def __init__(self, index):
        self.index = index
        self.dfsnum = 0
        self.visited = False
        self.succ = []  # list of edges to successors
        self.shortest = 0

    def print_node(self):
        print("Node with index {} and dfs {} and successors:".format(self.index, self.dfsnum))
        for succ in self.succ:
            print("\t{}".format(succ.node2.index))
        print()

    def print_shortest(self):
        print("Node with index {} has shortest path from root {}".format(self.index, self.shortest))

File: Lab6-RailwayPlanning/C++ Solution/node.cc
#include "node.h"
#include "edge.h"

using std::cout;
using std::endl;

Node::Node(const int& i) : index(i), visited(false), previous_node(nullptr), previous_edge(nullptr) {}

void Node::print_node(){
    cout << "Node " << index << std::boolalpha << ", visited=" << visited <<
            ", edges: ";
        for(const auto& e : edges){
            cout << "\n\t";
            e->print_edge();
        }
        cout << endl;
}